src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3762,9 +3836,14 @@ public override void VisitForEachIterationVariables(BoundForEachStatement node)
                     {
                         ReportWWarning(node.IterationVariableType.Syntax);
                     }
-                    isNullableIfReferenceType = result.IsNullable;
+                    annotation = result.NullableAnnotation;
+                }
+
+                int slot = GetOrCreateSlot(iterationVariable);
+                if (slot > 0)
```

Comment:
>if (slot > 0) [](start = 16, length = 13)

Please assert.

Reply:
> Please assert.

This cannot be replaced with an assert. It is valid for GetOrCreateSlot to return -1. For example, -1 is returned for an empty structure. This was caught by an existing test, somehow BitVector didn't throw for index -1.

---
In reply to: [231338348](https://github.com/dotnet/roslyn/pull/30913#discussion_r231338348) [](ancestors = 231338348)